tests/neg/i6565.scala

```diff
@@ -9,9 +9,9 @@ def (o: Lifted[O]) flatMap [O, U] (f: O => Lifted[U]): Lifted[U] = ???
 val error: Err = Err()
 
 lazy val ok: Lifted[String] = { // ok despite map returning a union
-  point("a").map(_ => if true then "foo" else error) // error
+  point("a").map(_ => if true then "foo" else error) // ok
 }
 
 lazy val bad: Lifted[String] = { // found Lifted[Object]
   point("a").flatMap(_ => point("b").map(_ => if true then "foo" else error)) // error
```

Comment:
I would still say that this should not be an error, the expected type of the body of `flatMap` is `T => Lifted[U]`, and as the test above now shows, `point("b").map(_ => if true then "foo" else error)` would unify with `Lifted[String]`

Reply:
Agreed, I left it as is because I couldn't manage to fix it.

Reply:
Ah I see, this is very nice